Gd 108

Barmalej24 opened the pull request.

Barmalej24 committed:
#GD-74 state: To Verify work 20m
Added Separate domain layer,
Domain entities have been added for settings and messages,
Added entity interface

Barmalej24 committed:
#GD-74 state: To Verify work 20m
Added Separate domain layer,
Domain entities have been added for settings and messages,
Added entity interface

Barmalej24 committed:
#GD-74 state: To Verify work 40m
Added Email ValueObject,
Added Exceptions for all domain class
Changed class constructors
Remove metods Create

Barmalej24 committed:
#GD-74 state : To Verify work 30м
Added BaseInterfaces

Barmalej24 committed:
#GD-74 state : To Verify work 30м
Added BaseInterfaces

Barmalej24 committed:
fix#GD-74 state: To Verify work 30м
Added BaseInterfaces

Barmalej24 committed:
#GD-74 state: To Verify work 30м
Added BaseInterfaces

Barmalej24 committed:
Fix

Barmalej24 committed:
fix

Barmalej24 committed:
#GD-74 state: To Verify
Change Exception
Remove All NotificationMicroservice.Domain

Barmalej24 committed:
#GD-74 state: To Verify
Change StringExceptions

Barmalej24 committed:
Editing XML documentation
Fixed models and interfaces

Barmalej24 committed:
#GD-109 state: To Verify work 90m
Editing XML documentation
Fixed models and interfaces

Barmalej24 committed:
#GD-111 state: To Verify work 120m
Added Tests for Domain Models and Exceptions
Fix Exceptions

Barmalej24 committed:
#GD-109 state: To Verify work 20m
Added Documentation for Exseptions

Barmalej24 committed:
#GD-109 state: To Verify work 30m
Rename Models
Add new ValueObject - Connection (address, port)

Barmalej24 committed:
fix REGEX string server address

Barmalej24 committed:
#GD-111 state: To Verify work 120m
Fixed Tests for Domain Models and Exception

oleggolen commented on EmailSenderMicroservice.Domain/Entity/Message.cs:
```diff
@@ -0,0 +1,91 @@
+        public Message(Guid id, string email, string messageType, string messageText, bool status, DateTime createDate)
+        {
+
+            if (id == Guid.Empty)
+            {
+                throw new MessageGuidEmptyException(ExceptionStrings.ERROR_ID, id.ToString());
+            }
+
+            if (string.IsNullOrEmpty(messageType))
+            {
+                throw new MessageTypeNullOrEmptyException(ExceptionStrings.ERROR_TYPE, messageType);
+            }
+
+            if (string.IsNullOrEmpty(messageText))
+            {
+                throw new MessageTextNullOrEmptyException(ExceptionStrings.ERROR_TEXT, messageText);
+            }
```
Проверить на строку содержащую просто пробел

oleggolen commented on EmailSenderMicroservice.Domain/Entity/Setting.cs:
```diff
@@ -0,0 +1,83 @@
+                throw new SettingGuidEmptyException(ExceptionStrings.ERROR_ID, id.ToString());
+            }
+
+            if (string.IsNullOrEmpty(password))
+            {
+                throw new SettingPasswordNullOrEmptyException(ExceptionStrings.ERROR_SERVER_PASS, password.ToString());
+            }
```
Проверить на строку содержащую просто пробел

oleggolen commented on EmailSenderMicroservice.Domain/Exception/Message/MessageGuidEmptyException.cs:
```diff
@@ -0,0 +1,35 @@
+    /// <summary>
+    /// Исключение пустого значения идентификатора
+    /// </summary>
+    public class MessageGuidEmptyException : ArgumentException
```
Выбрать более конкретное исключение ArgumentNullException

oleggolen commented on EmailSenderMicroservice.Domain/Exception/Message/MessageGuidEmptyException.cs:
```diff
@@ -0,0 +1,35 @@
+            : base(message, innerException)
+        {
+        }
+    }
```
Убрать неиспользуемые конструкторы

oleggolen commented on EmailSenderMicroservice.Domain/Exception/Message/MessageTextNullOrEmptyException.cs:
```diff
@@ -0,0 +1,35 @@
+        public MessageTextNullOrEmptyException(string? message, ArgumentException? innerException) 
+            : base(message, innerException)
+        {
+        }
```
Убрать не используемые конструкторы, и убрать nullable у параметров

oleggolen commented on EmailSenderMicroservice.Domain/Exception/Message/MessageTextNullOrEmptyException.cs:
```diff
@@ -0,0 +1,35 @@
+    /// <summary>
+    /// Исключение пустого значения текста сообщения
+    /// </summary>
+    public class MessageTextNullOrEmptyException : ArgumentException
```
Выбрать базовым более конкретное исключение ArgumentNUllException

oleggolen commented on EmailSenderMicroservice.Domain/Exception/Message/MessageTypeNullOrEmptyException.cs:
```diff
@@ -0,0 +1,35 @@
+        /// </summary>
+        /// <param name="message">Информационное сообщение</param>
+        /// <param name="value">значение параметра вызвашвего исключение</param>
+        public MessageTypeNullOrEmptyException(string? message, string? value)
```
Убрать неиспользуемые конструкторы, и в параметрах убрать nullable

oleggolen commented on EmailSenderMicroservice.Domain/Exception/Message/MessageTypeNullOrEmptyException.cs:
```diff
@@ -0,0 +1,35 @@
+    /// <summary>
+    /// Исключение пустого значения типа сообщения
+    /// </summary>
+    public class MessageTypeNullOrEmptyException : ArgumentException
```
Выбрать более конкретный тип исключения ArgumentNullException

oleggolen commented on EmailSenderMicroservice.Domain/Exception/Resources/ExceptionStrings.cs:
```diff
@@ -0,0 +1,18 @@
+﻿namespace EmailSenderMicroservice.Domain.Exception.Resources
+{
+    public class ExceptionStrings
```
Сделать класс статическим

oleggolen commented on EmailSenderMicroservice.Domain/Exception/Setting/SettingGuidEmptyException.cs:
```diff
@@ -0,0 +1,35 @@
+    /// <summary>
+    /// Исключение пустого значения идентификатора
+    /// </summary>
+    public class SettingGuidEmptyException : ArgumentException
```
Выбрать более конкретный тип исключения ArgumentNullException

oleggolen commented on EmailSenderMicroservice.Domain/Exception/Setting/SettingGuidEmptyException.cs:
```diff
@@ -0,0 +1,35 @@
+        }
+
+        /// <summary>
+        /// Конструктор с информационным сообщением и значением параметра
```
Убрать неиспользуемые конструкторы

oleggolen commented on EmailSenderMicroservice.Domain/Exception/Setting/SettingLoginNullOrEmptyException.cs:
```diff
@@ -0,0 +1,35 @@
+    /// <summary>
+    /// Исключение пустого значения логина
+    /// </summary>
+    public class SettingLoginNullOrEmptyException : ArgumentException
```
Выбрать более конкретное исключение ArgumentNullException

oleggolen commented on EmailSenderMicroservice.Domain/Exception/Setting/SettingPasswordNullOrEmptyException.cs:
```diff
@@ -0,0 +1,35 @@
+        /// <summary>
+        /// Основной конструктор
+        /// </summary>
+        public SettingPasswordNullOrEmptyException()
```
Убрать неиспользуемые конструкторы и сделать параметры не nulable

oleggolen commented on EmailSenderMicroservice.Domain/Exception/Setting/SettingLoginNullOrEmptyException.cs:
```diff
@@ -0,0 +1,35 @@
+        /// Основной конструктор
+        /// </summary>
+        public SettingLoginNullOrEmptyException()
+        {
```
Убрать неиспользуемые конструкторы и сделать параметры не nullable

oleggolen commented on EmailSenderMicroservice.Domain/Exception/Setting/SettingServerAddressLengthException.cs:
```diff
@@ -0,0 +1,35 @@
+    {
+        /// <summary>
+        /// Основной конструктор
+        /// </summary>
```
Убрать неиспользуемые конструкторы и сделать параметры не nullable

oleggolen commented on EmailSenderMicroservice.Domain/Exception/Setting/SettingServerAddressNullOrEmptyException.cs:
```diff
@@ -0,0 +1,35 @@
+    /// <summary>
+    /// Исключение пустого значения адреса сервера
+    /// </summary>
+    public class SettingServerAddressNullOrEmptyException : ArgumentException
```
Сделать более конкретный тип исключения ArgumentNullException

oleggolen commented on EmailSenderMicroservice.Domain/Exception/Setting/SettingServerAddressNullOrEmptyException.cs:
```diff
@@ -0,0 +1,35 @@
+    /// </summary>
+    public class SettingServerAddressNullOrEmptyException : ArgumentException
+    {
+        /// <summary>
```
Убрать неиспользуемые конструторы и сделать параметры не nullable

oleggolen commented on EmailSenderMicroservice.Domain/Exception/Setting/SettingServerPortException.cs:
```diff
@@ -0,0 +1,35 @@
+    /// <summary>
+    /// Исключение значения параметра порта сервера
+    /// </summary>
+    public class SettingServerPortException : ArgumentException
```
Тут лучше использовать FormatExcepton, так как порт - это значение, представляемое целым числом в диапозоне 0 65535, всё остальное - не порт, то есть жёстко задан формат, и его нарушение есть FormatException

oleggolen commented on EmailSenderMicroservice.Domain/Exception/Setting/SettingServerPortException.cs:
```diff
@@ -0,0 +1,35 @@
+        /// <summary>
+        /// Основной конструктор
+        /// </summary>
+        public SettingServerPortException()
```
Убрать неиспользуемые конструкторы и сделать параметры не nullable

oleggolen commented on EmailSenderMicroservice.Domain/Exception/ValueObject/EmailInvalidException.cs:
```diff
@@ -0,0 +1,35 @@
+    /// <summary>
+    /// Исключение проверки строки на соответсвуе Email
+    /// </summary>
+    internal class EmailInvalidException : ArgumentException
```
Тут лучше использовать FormatException, так как Email имеет очень чётко заранее заданный формат, и его нарушение приводит к невалидному Email

oleggolen commented on EmailSenderMicroservice.Domain/Exception/ValueObject/EmailInvalidException.cs:
```diff
@@ -0,0 +1,35 @@
+    /// </summary>
+    internal class EmailInvalidException : ArgumentException
+    {
+        /// <summary>
```
Убрать не используемые конструкторы и сделать параметры не nullable

oleggolen commented on EmailSenderMicroservice.Domain/Interface/Model/IEntity.cs:
```diff
@@ -0,0 +1,19 @@
+﻿namespace EmailSenderMicroservice.Domain.Interface.Model
```
Лучше перенести в проект или сделать подпапку в Entities EmailSenderMicroservice.Domain.Entities.Base

oleggolen commented on EmailSenderMicroservice.Domain/Interface/Model/IEntity.cs:
```diff
@@ -0,0 +1,19 @@
+    /// Описание базовой Entity 
+    /// </summary>
+    /// <typeparam name="TKey"> Тип первичного ключа. </typeparam>
+    public interface IEntity<Tkey>
```
Добавить проверку на то, что ключ является примитивным типом для корректной работы базы данных

oleggolen commented on EmailSenderMicroservice.Domain/Interface/Repository/IBaseRepository.cs:
```diff
@@ -0,0 +1,32 @@
+    /// <summary>
+    /// Описания методов для базового репозитория.
+    /// </summary>
+    public interface IBaseRepository<TEntity, TKey>
```
Сделать проверку на типы, что TEntity является классом, реализующим интерфейс IEntity, а TKEy - объектом примитивного типа для корректной работы базы данных

oleggolen commented on EmailSenderMicroservice.Domain/Interface/Repository/IBaseRepository.cs:
```diff
@@ -0,0 +1,32 @@
+﻿namespace EmailSenderMicroservice.Domain.Interface.Repository
```
Вынести в проект EmailSenderMicroservice.Doamin.Repositories.Abstractions

oleggolen commented on EmailSenderMicroservice.Domain/Interface/Repository/IMessageRepository.cs:
```diff
@@ -0,0 +1,11 @@
+﻿using EmailSenderMicroservice.Domain.Models;
+
+namespace EmailSenderMicroservice.Domain.Interface.Repository
```
Вынести в проект EmailSenderMicroservice.Repositories.Abstractions

oleggolen commented on EmailSenderMicroservice.Domain/Interface/Repository/ISettingRepository.cs:
```diff
@@ -0,0 +1,25 @@
+﻿using EmailSenderMicroservice.Domain.Models;
+
+namespace EmailSenderMicroservice.Domain.Interface.Repository
```
Вынести в проект EmailSenderMicroservice.Repositories.Abstractions

oleggolen commented on EmailSenderMicroservice.Domain/Interface/Service/IBaseService.cs:
```diff
@@ -0,0 +1,29 @@
+﻿namespace EmailSenderMicroservice.Domain.Interface.Service
```
Вынести из домена в проект EmailSenderMicroservice.Applicaton.Services.Abstractions

oleggolen commented on EmailSenderMicroservice.Domain/Interface/Service/IBaseService.cs:
```diff
@@ -0,0 +1,29 @@
+    /// <summary>
+    /// Описания методов для базового сервиса.
+    /// </summary>
+    public interface IBaseService<TEntity, TKey>
```
Добавить проверки на то, что TEntity наследуется от базового типа сущности слоя приложения или реализует интерфейс IEntity, а TKey - примитивный тип для корректной работы базы данных

oleggolen commented on EmailSenderMicroservice.Domain/Interface/Service/IMessageService.cs:
```diff
@@ -0,0 +1,12 @@
+﻿using EmailSenderMicroservice.Domain.Models;
+
+namespace EmailSenderMicroservice.Domain.Interface.Service
```
Перенести в EmailSenderMicroservice.Application.Services.Abstractions или если это доменный сервис то в EmailSenderMicroservice.Doamin.Services.Abstractions

oleggolen commented on EmailSenderMicroservice.Domain/Interface/Service/ISettingService.cs:
```diff
@@ -0,0 +1,16 @@
+﻿using EmailSenderMicroservice.Domain.Models;
+
+namespace EmailSenderMicroservice.Domain.Interface.Service
```
Перенести в EmailSenderMicroservice.Application.Services.Abstractions или если это доменный сервис то в EmailSenderMicroservice.Doamin.Services.Abstractions

oleggolen reviewed (changes requested):
Код написан хорошо, грамотно, очень хорошо написан базовый набор тестов, но стоит обратить внимание на следующие:
1. Расположение классов в пространствах имён в соответствии с шаблоном архитектуры
2. Базовые типы исключений должны быть более близкими и конкретными
3. Стоит сделать иерархию исключений, так как многие исключения себя повторяют и очень похожи по структуре
4. Добавиь проверки в ValueObject на пробельные строки
5. В домене сущности называются Entities а не Models
Очень надеюсь, что полученные рекомендации помогут следовать структуре трёхслойной архитектуры, писать хорошо документированный , понятный и читаемый код с использованием приципов DDD

Barmalej24 committed:
#GD-122 state: To Progress work 1h 50m
Removed unused constructors from exceptions
Changed inherited exceptions for null exceptions
Changed the entity structure
Made changes to namespice
Deleted abstraction

Barmalej24 committed:
#GD-122 state: To Verify work 1h 40m
Removed unused constructors from exceptions
Changed inherited exceptions for null exceptions
Changed the entity structure
Made changes to namespice
Removed interfaces from the domain
Modify Test

Barmalej24 committed:
modifed exceptions
